src/assets/Zombie.java

```diff
@@ -185,4 +200,37 @@ public void attack(Board board) {
 			board.removePlant(getRow(), getCol());
 		}
 	}
+	
+	/**
+	 * Restore the zombie's original speed. 
+	 */
+	public abstract void restoreSpeed();

```

Comment:
The reason why this is abstract is because subclasses of Zombie has different speed values and the values are private constants.  Which makes me think why are default attributes private in the first place?